tests/util/rest.go

```diff
@@ -167,3 +168,17 @@ func RetrieveRunID(t *testing.T, responseData []byte) string {
 	}
 	return runResponse.RunID
 }
+
+func ApplyPipelineYAML(t *testing.T, yamlPath, namespace string) {
+	// Validate inputs to prevent command injection
+	if strings.ContainsAny(yamlPath, "|;&$`") || strings.ContainsAny(namespace, "|;&$`") {
+		t.Fatalf("Invalid characters in yamlPath or namespace")
+	}
+	if !strings.HasSuffix(yamlPath, ".yaml") && !strings.HasSuffix(yamlPath, ".yml") {
+		t.Fatalf("yamlPath must be a YAML file")
+	}
```

Comment:
it feels somewhat clunky to do (and maintain) our own input validation here, especially since I don't believe this is ever exposed to a public interface.  Perhaps a quick filepath check with a built-in library like `os` would be better here?